Registration/RegisterWithAzure.psm1

```diff
@@ -9,6 +9,45 @@ You must also have access to an account / directory that is an owner or contribu
 
 #>
 
+function Initialize-AzEnvironment{
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$false)]
+        [String] $Name = 'CustomCloud',
+
+        [Parameter(Mandatory=$true, ParameterSetName='CloudManifest')]
+        [String] $CloudManifestFilePath,
+
+        [Parameter(Mandatory=$true, ParameterSetName='CloudARMEndpoint')]
+        [String] $CloudARMEndpoint
+    )
+
+    if ( $PSCmdlet.ParameterSetName -eq 'CloudManifest' ){
+        $cloudJson = Get-Content $CloudManifestFilePath -Raw | ConvertFrom-Json
+        $cloudJsonContent = $cloudJson.DeploymentData
+        $CloudARMEndpoint = $cloudJsonContent.CustomEnvironmentEndpoints.CustomCloudARMEndpoint
+    }
+
+    $fullUri = $CloudARMEndpoint.TrimEnd('/')+"/metadata/endpoints?api-version=2015-01-01"
```

Comment:
You need not query the metadata endpoint, instead use -ArmEndpoint

Reply:
I remember -ArmEndpoint method was giving error for EX cloud with some like unknown 'ResourceManagerEndpoint'. Also, can we set 'EnableAdfsAuthentication = $true' with -ARMEndpoint method?

Reply:
Ah , i see. You cannot set EnableAdfsAuthentication with the -ArmEdnpoint parameter set, You can do it Set-AzEnvironment after doing Add

What you have is fine if we need to set EnableAdfsAuthentication